logging/google/cloud/logging/handlers/transports/background_thread.py

```diff
@@ -19,137 +19,224 @@
 
 import atexit
 import copy
+import logging
 import threading
 
+from six.moves import queue
+
 from google.cloud.logging.handlers.transports.base import Transport
 
-_WORKER_THREAD_NAME = 'google.cloud.logging.handlers.transport.Worker'
+_DEFAULT_GRACE_PERIOD = 5.0
+_DEFAULT_BATCH_SIZE = 10
+_WORKER_THREAD_NAME = 'google.cloud.logging.Worker'
+_WORKER_TERMINATOR = object()
+_LOGGER = logging.getLogger(__name__)
 
 
-class _Worker(object):
-    """A threaded worker that writes batches of log entries
+def _get_many(q, max=None):
+    """Get multiple items from a Queue.
+
+    Gets at least one (blocking) and at most :param:`max` items (non-blocking)
+    from a given Queue. Does not mark the items as done.
 
-    Writes entries to the logger API.
+    :type q: ~queue.Queue
+    :param q: The Queue to get items from.
 
-    This class reuses a single :class:`Batch` method to write successive
-    entries.
+    :type max: int
```

Comment:
This is a real nit, but  hesitate using max as a variable name it shadows a builtin.